### PR TITLE
fix the issues in deploy-operator.md

### LINKED ISSUE #75: Following docs/deploy-operator.md does not result in successful installation due to missing / outdated content
As a first time user, I've been trying to install the latest `master` version of the operator on my local machine (in KinD) for testing purposes. However, after following the instructions in `docs/deploy-operator.md`, I was not able to finish the installation successfully without undertaking further steps. In particular, the issues I discovered:

- Applying `example/manifest/observatorium.yaml` results in validation error:
  ```
  error: error validating "https://raw.githubusercontent.com/observatorium/operator/master/example/manifests/observatorium.yaml": error validating data: [ValidationError(Observatorium.spec): unknown field "apiQuery" in io.observatorium.core.v1alpha1.Observatorium.spec, ValidationError(Observatorium.spec.securityContext): unknown field "fsGroup" in io.observatorium.core.v1alpha1.Observatorium.spec.securityContext]; if you choose to ignore these errors, turn validation off with --validate=false
  ```
  since it includes unknown fields `apiQuery` (removed in https://github.com/observatorium/operator/pull/9) and `fsGroup` (which does not appear to be included in `securityContext`, but rather in `podSecurityContext`); a possible resolution via https://github.com/observatorium/operator/pull/66.

- Observatorium API pod fails to start due to missing secret `observatorium-xyz-tls` and missing configmaps `observatorium-xyz-tls` and `test-ca-tls` - these are not created in the process / user is not instructed to create these.

-  The [example CR](https://github.com/observatorium/operator/blob/2090181136dbc0b6468da3ae22fb6858ee5da9fd/example/manifests/observatorium.yaml#L40) seems to expect Dex to be available as an issuer, however user is not instructed to deploy it.

- For post-installation test, the observatorium up metrics job has wrong manifest link (should be https://raw.githubusercontent.com/observatorium/observatorium/main/configuration/tests/manifests/observatorium-up-metrics.yaml instead of https://raw.githubusercontent.com/observatorium/deployments/master/tests/manifests/observatorium-up-metrics.yaml). 

If possible, these issues should be addressed and content updated, to make the deployment instructions clearer for users.

### PR DESCRIPTION
**Pull request by clyang82**
fixed: https://github.com/observatorium/operator/issues/75

Signed-off-by: clyang82 <chuyang@redhat.com>

### CODE REVIEW

**Comment by clyang82** [comment]
/hold
I will update once https://github.com/observatorium/observatorium/pull/452 is merged

File: docs/deploy-operator.md [hunk 1]
```diff
@@ -25,16 +24,31 @@ kubectl create namespace observatorium
...
+kubectl apply -f https://raw.githubusercontent.com/observatorium/observatorium/main/configuration/examples/dev/manifests/minio-secret-loki.yaml
 kubectl apply -f https://raw.githubusercontent.com/observatorium/observatorium/main/configuration/examples/dev/manifests/minio-secret-thanos.yaml
 kubectl apply -f https://raw.githubusercontent.com/observatorium/observatorium/main/configuration/examples/dev/manifests/minio-pvc.yaml
 kubectl apply -f https://raw.githubusercontent.com/observatorium/observatorium/main/configuration/examples/dev/manifests/minio-deployment.yaml
 kubectl apply -f https://raw.githubusercontent.com/observatorium/observatorium/main/configuration/examples/dev/manifests/minio-service.yaml
 ```
 
+#### Dex
+
+For **testing purposes** you may use [dex](https://github.com/dexidp/dex) which is an identity service for observatorium API as described below.
```
Comment (squat) [inline docs/deploy-operator.md:40]: small nit:
```suggestion
For **testing purposes** you may use [dex](https://github.com/dexidp/dex) as an OIDC identity provider for the Observatorium API as described below.
```

File: docs/deploy-operator.md [hunk 2]
```diff
@@ -68,6 +80,12 @@ kubectl apply -f https://raw.githubusercontent.com/observatorium/operator/master
 kubectl apply -n observatorium -f https://raw.githubusercontent.com/observatorium/operator/master/example/manifests/observatorium.yaml
 ```
 
+For **testing purposes**, you may use the certificate and key as described below.
```
Comment (squat) [inline docs/deploy-operator.md:83]:
```suggestion
For **testing purposes**, you may use the end-to-end test certificate and key as described below.
```

File: example/main.jsonnet [hunk 3]
```diff
@@ -171,7 +166,6 @@ local dex = (import 'github.com/observatorium/observatorium/configuration/compon
       volumeClaimTemplate: obs.loki.config.volumeClaimTemplate,
     },
     securityContext: {
-      fsGroup: 65534,
```
Comment (squat) [inline example/main.jsonnet:174]: why do we remove this?

**Review by squat — COMMENTED** [review]
Looks great @clyang82. I have one doubt and two small suggestions

File: example/main.jsonnet (hunk 3 above)
Comment (squat) [reply example/main.jsonnet:174]: ping @clyang82

Comment (clyang82) [reply example/main.jsonnet]: It should not be removed. In order to fix the problem `unknown field "fsGroup" in io.observatorium.core.v1alpha1.Observatorium.spec.securityContext`, I need to update api to use `PodSecurityContext`

**Review by squat — APPROVED** [review]
thanks @clyang82

**Pull request merged**